set-dyn-ip.hs

```diff
@@ -82,4 +86,15 @@ changeIpAddr zone host externIp = do
 perReqDelay :: Int
 perReqDelay = 1000000
 
+getCurrentIpAddr :: String -> String -> IO (Maybe String)
+getCurrentIpAddr zone host = do
+    env <- newEnv (FromEnv (toText "AWS_ACCESS_KEY")
+                           (toText "AWS_SECRET_KEY")
+                           Nothing
+                           Nothing)
```

Comment:
this `newEnv` block is repeated here and in `changeIpAddr`; please extract it to a helper function.